Resize should work within Grid details row (#7341)

Ansku opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/framework/9808)
<!-- Reviewable:end -->

Ansku committed:
Resize should work within Grid details row (#7341)

tsuoanttila commented:
When fixing this to match the 7.7 pull request, please take manually run the 4 tests that failed in the validation build.

Ansku committed:
Further tweaks to details row resize handling (#7341)

- comments
- logic fix to ensure listeners are removed correctly in
non-compatibility version as well
- further testing & minor test tweaks

Ansku committed:
Test tweaks for details row resizing (#7341)

- Expanded allowed height difference since details row resizing working
at all is more important than keeping it pixel perfect.
- Gave up on testing the multi-row resize for PhantomJS since no
workaround got the initial button click working as it's supposed to.

ahie commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 8 files at r1, 6 of 6 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9808#-:-Ks7au0bVAC87AfvMxLu:bnfp4nl)*
<!-- Sent from Reviewable.io -->
